### PR TITLE
feat: add a .devcontainer config for VSCode development

### PR DESCRIPTION
**Pull request by disposedtrolley**
Adds a .devcontainer config to enable a fast-path local development
experience for VSCode users. This uses Docker to provision standard
development environments with all of the system dependencies required
for NodeJS development. The repository is mounted within the container,
so changes made to the code are reflected back on the host system.

It also configures VSCode with some sensible default extensions (ESLint,
Prettier, Docker).

See https://code.visualstudio.com/docs/remote/containers#_getting-started
for more information.

### CODE REVIEW

File: .devcontainer/scripts/start.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+# Performs setup tasks _after_ the .devcontainer Docker container is created.
+
+# Install Node dependencies.
+yarn install
```
Comment (ashiquem) [inline .devcontainer/scripts/start.sh:6]: would we need to install `yarn` in the container first? not clear if the base image comes with it @disposedtrolley

Comment (disposedtrolley) [reply .devcontainer/scripts/start.sh:6]: It comes with it! I think MS graciously installs it as part of the NodeJS base image for dev containers haha

**Review by ashiquem — APPROVED** [review]
great job, works nicely on ubuntu x86

**Review by awpala — APPROVED** [review]

**Review by ctran4347 — APPROVED** [review]
I got this working on Mac OSX 12.3.1 Monterey. This looks good!

**Pull request merged**